data/virt_autotest/fetch_logs_from_guest.sh

```diff
@@ -0,0 +1,260 @@
+function fetch_logs_from_guest() {
+	local guest_domain=$1
+	local guest_ipaddr=$2
+	local logs_folder=$3
+	shift
+	shift
+	shift
+	local extra_logs=($@)
+
+	local ret1=0
+	fetch_logs_from_guest_via_ssh ${guest_domain} ${guest_ipaddr} ${logs_folder}
+	if [[ $? -ne 0 ]];then
+	   echo -e "Try to use libguestfs tools to fetch ${logs_folder} from ${guest_domain}."
+           fetch_logs_from_guest_via_libguestfS ${guest_domain} ${logs_folder} ${logs_folder}
+	   ret1=$?
+	fi
+
+	local ret2=0
+	if [[ ${extra_logs[@]} != "" ]];then
+           echo -e "Try to use libguestfs tools to fetch ${extra_logs[@]} from ${guest_domain}."
+           for eachlog in ${extra_logs[@]};do
+	       fetch_logs_from_guest_via_libguestfS ${guest_domain} ${eachlog} ${logs_folder}
+	       ret2=$(( ${ret2} | $? ))
+	   done           
+	   if [[ ${ret2} -eq 0 ]];then
+	      echo -e "Copied out ${extra_logs[@]} from ${guest_domain} successfully."
+	   else
+	      echo -e "Failed to copy out ${extra_logs[@]} from ${guest_domain}."
+	   fi
+	fi
+        
+	virsh start ${guest_domain}
+	if [[ ${ret1} -eq 0 && ${ret2} -eq 0 ]];then
+	   return 0
+	else
+	   return 1
+	fi
+}
+
+#Compress logs folder on local host which contains all logs from host and guest
+function compress_virt_logs_folder() {
+	local logs_folder=$1
+	local logs_root=${logs_folder/\//}
+        logs_root=${logs_root/\/*/}	
+
+	echo -e "tar -czvf /${logs_root}/virt_logs_all.tar.gz ${logs_folder}"
+        tar -czvf /${logs_root}/virt_logs_all.tar.gz ${logs_folder}
+	if [[ $? -eq 0 ]];then
+           echo -e "Successfully compressed ${logs_folder} to /${logs_root}/virt_logs_all.tar.gz"
+	   return 0
+       else
+           echo -e "Failed to ${logs_folder} to /${logs_root}/virt_logs_all.tar.gz"
+	   return 1
+	fi
+}
+
+#Usage and help info for the script
+help_usage(){
+	echo "script usage: $(basename $0) [-f \"Logs folder which contains logs collected(Can be omitted/Default to /tmp/virt_logs_residence)\"] [-g \"guests to be involved, for example, \"guest1 guest2 guest3\" or all or none(Can be omitted/Default to all)\"] [-e \"Extra folders or files to be fetched from guest, for example, \"log_file1 log_file2 log_folder1\"(Can be omitted/Default to nothing)\"] [-h help]"
+}
+
+fetch_logs_from_guest_log="/var/log/fetch_logs_from_guest.log"
+virt_guests_wanted=""
+virt_logs_folder=""
+virt_extra_logs_guest=""
+fetch_logs_from_guest_log_result=0
+rm -f -r ${fetch_logs_from_guest_log}
+
+#Parse input arguments
+while getopts 'l:g:e:h' OPTION; do
+  case "$OPTION" in
+    f)
+      virt_logs_folder="$OPTARG"
+      echo "Logs folder is ${virt_logs_folder}" | tee -a ${fetch_logs_from_guest_log}
+      ;;
+    g)
+      virt_guests_wanted="$OPTARG"
+      echo "The guests involved are ${virt_guests_wanted}" | tee -a ${fetch_logs_from_guest_log}
+      ;;
+    e)
+      virt_extra_logs_guest="$OPTARG"
```

Comment:
what if extra logs host?

Reply:
That already happens in virt_logs_collector.sh

Reply:
You are right.